src/addrman.h

```diff
@@ -523,9 +525,10 @@ friend class CAddrManTest;
             s >> serialized_asmap_version;
         }
 
-        for (int n = 0; n < nNew; n++) {
-            CAddrInfo &info = mapInfo[n];
-            int bucket = entryToBucket[n];
+        for (auto bucket_entry : bucket_entries) {
```

Comment:
I think `bucket_entries.size() >= nNew`, let me use an example:

```
mapInfo[id=3] = 1.1.1.1
// for the example, I pick up some arbitrary indexes in vvNew
vvNew[2][5] = id=3
vvNew[4][9] = id=3
nNew = 1
```

This would be serialized on disk as:

```
1 (nNew)
0 (nTried)
1024 (new_bucket_count)
1.1.1.1 (list of new addresses) (*)
(zero tried addresses)
0 (number of elements in vvNew[0])
0 (number of elements in vvNew[1])
1 (number of elements in vvNew[2])
0 (the index in (*) of the first non empty element in vvNew[2])
0 (number of elements in vvNew[3])
1 (number of elements in vvNew[4])
0 (the index in (*) of the first non empty element in vvNew[4])
0 (number of elements in vvNew[5])
...
0 (number of elements in vvNew[1023])
```

During unserialization, in the new code from this PR, this will result in:

```
bucket_entries[0] = (bucket = 2, index in (*) = 0)
bucket_entries[1] = (bucket = 4, index in (*) = 0)
```

So, `bucket_entries.size() = 2` while `nNew = 1`.

Reply:
This was part of the bug that this PR fixes. nNew is the total number of new addrs. Each new addr can appear in more than one new bucket, so the number of entries in bucket_entries could be higher than nNew. The old code was potentially throwing away these additional placements.